src/main/kotlin/app/revanced/patches/all/screenshot/removerestriction/patch/RemoveScreenshotRestrictionPatch.kt

```diff
@@ -1,35 +1,37 @@
 package app.revanced.patches.all.screenshot.removerestriction.patch
 
-import app.revanced.extensions.findMutableMethodOf
 import app.revanced.patcher.annotation.Description
 import app.revanced.patcher.annotation.Name
 import app.revanced.patcher.annotation.Version
-import app.revanced.patcher.data.BytecodeContext
 import app.revanced.patcher.extensions.replaceInstruction
-import app.revanced.patcher.patch.BytecodePatch
-import app.revanced.patcher.patch.PatchResult
-import app.revanced.patcher.patch.PatchResultSuccess
 import app.revanced.patcher.patch.annotations.Patch
 import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import app.revanced.util.patch.AbstractTransformInstructionsPatch
 import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.iface.ClassDef
+import org.jf.dexlib2.iface.Method
+import org.jf.dexlib2.iface.instruction.Instruction
 import org.jf.dexlib2.iface.instruction.formats.Instruction35c
 import org.jf.dexlib2.iface.reference.MethodReference
+import java.util.*
+
+private typealias InstructionInfo = Triple<RemoveScreenshotRestrictionPatch.MethodCall, Instruction35c, Int>
 
 @Patch(false)
 @Name("remove-screenshot-restriction")
 @Description("Removes the restriction of making screenshots.")
 @Version("0.0.1")
-class RemoveScreenshotRestrictionPatch : BytecodePatch() {
+class RemoveScreenshotRestrictionPatch : AbstractTransformInstructionsPatch<InstructionInfo>() {
 
     private companion object {
         const val INTEGRATIONS_CLASS_DESCRIPTOR = "Lapp/revanced/all/screenshot/removerestriction/RemoveScreenshotRestrictionPatch;"
     }
 
     // Information about method calls we want to replace
-    private enum class MethodCall(
+    enum class MethodCall(
```

Comment:
This enum can be internal to the package, since it is now necessary to access its fields.

Reply:
Won’t work because of
```kotlin
private typealias InstructionInfo = Triple<RemoveScreenshotRestrictionPatch.MethodCall, Instruction35c, Int>
class RemoveScreenshotRestrictionPatch : AbstractTransformInstructionsPatch<InstructionInfo>() {
```

Reply:
That's odd. The type alias is private so it should've been possible to make `MethodCall` private.

Reply:
> private means that the member is visible inside this class only (including all its members).

https://kotlinlang.org/docs/visibility-modifiers.html#class-members